kmir/src/kmir/__main__.py

```diff
@@ -166,16 +284,83 @@ def create_argument_parser() -> ArgumentParser:
         help='Generate a haskell-backend bug report for the execution',
     )
     run_subparser.add_argument(
-        '--max-depth',
+        '--depth',
         default=None,
         type=int,
-        help='Stop execution after `max-depth` rewrite steps',
+        help='Stop execution after `depth` rewrite steps',
+    )
+
+    # Prove
+    prove_subparser = command_parser.add_parser(
+        'prove', parents=[logging_args], help='Prove a MIR specification WARN: EXPERIMENTAL AND WORK IN PROGRESS'
+    )
+    prove_subparser.add_argument(
+        '--definition-dir',
+        dest='definition_dir',
+        type=dir_path,
+        help='Path to LLVM definition to use.',
+    )
+    prove_subparser.add_argument(
+        '--haskell-dir',
+        dest='haskell_dir',
+        type=dir_path,
+        help='Path to Haskell definition to use.',
+    )
+    prove_subparser.add_argument(
+        '--spec-file',
+        dest='spec_file',
+        type=file_path,
+        help='Path to specification file',
+    )
+    prove_subparser.add_argument(
+        '--bug-report',
+        action='store_true',
+        default=False,
+        help='Generate a haskell-backend bug report for the execution',
+    )
+    prove_subparser.add_argument(
+        '--depth',
+        default=None,
+        type=int,
+        help='Stop execution after `depth` rewrite steps',
+    )
+    prove_subparser.add_argument(
+        '--smt-timeout',
+        dest='smt_timeout',
+        type=int,
+        default=125,
+        help='Timeout in ms to use for SMT queries',
+    )
+    prove_subparser.add_argument(
+        '--smt-retry-limit',
+        dest='smt_retry_limit',
+        type=int,
+        default=4,
+        help='Number of times to retry SMT queries with scaling timeouts.',
+    )
+    prove_subparser.add_argument(
+        '--trace-rewrites',
+        default=False,
+        action='store_true',
+        help='Log traces of all simplification and rewrite rule applications.',
+    )
+    prove_subparser.add_argument(
+        '--save-directory',
+        dest='save_directory',
+        type=dir_path,
+        help='Path to KCFG proofs directory, directory must already exist.',
```

Comment:
I am not sure how `kevm` does this, but somehow they do not need the directory to already exist.